### ISSUE TITLE
Bring browser window to the front and keep it in focus

### ISSUE THREAD

**Issue by dariusf** [HubTurbo/HubTurbo#240]
Currently we can't find a way to do this using Selenium.

**Comment by damithc** [comment]
@dariusf @rd1992 Anybody working on this?

**Comment by dariusf** [comment]
Yes, @rd1992.

**Comment by rd1992** [comment]
will fix after class today

**Comment by rd1992** [comment]
@damithc Have a few clarifications. When should the browser be in front? Only when new issue needs to be created? and also how does the user come back to the HT window? He will have manually focus on the window to make it operational again

**Comment by damithc** [comment]
Ideally, whenever the user is working in HT, the bView should come to the
front (but not necessarily in focus)
Once the bView is given focus (e.g. to enter detail of a new issue), user
has to Ctrl+tab to get the focus back to HT. That's an extra step, but I'm
not sure if there's a way to avoid that.

**Comment by rd1992** [comment]
*"Ideally, whenever the user is working in HT, the bView should come to the
front (but not necessarily in focus)"* 
this already happens. the HT window and chrome window are stacked side by side with focus on HT.
On clicking new issue the bView loads new issue page but HT still has keyboard focus. Now the user can either press ALT + TAB or (click on the browser window) to go to the bView. Just to be clear you are trying to eliminate this ALT + TAB functionality by giving the focus to bView only when user wants to enter a new issue?

**Comment by damithc** [comment]
That's not what I meant. Assume both HT and bView are hidden behind other
windows. If I click on the HT window, the bView should come to the front as
well. The current release version doesn't work that way. I have to click on
both windows to get both of them to the front.

**Comment by rd1992** [comment]
ohh. ok i understand. thanks for the clarification.

**Comment by rd1992** [comment]
This seems like a bug with chromedriver. I can do everything with the window **except** bring focus to it. Tried the solution mentioned in the stackoverflow question but doesnt work. Tried executing window.focus() using javascript but that does not work in chrome. Trying it with firefox driver before i file an issue with the chromedriver developer

**Comment by damithc** [comment]
This one is a tough problem, but we need to solve it somehow. May be an
OS-specific solution?

**Comment by rd1992** [comment]
Yup trying an OS specific solution now

**Comment by rd1992** [comment]
Workaround is to use JNA to invoke native methods.

**Comment by damithc** [comment]
can do for both Windows and OS-X?

**Comment by damithc** [comment]
Also, make sure HT continues to work even if this feature does not work for
whatever reason.

**Comment by damithc** [comment]
Any further progress. I can't wait for this feature to come out.

**Comment by rd1992** [comment]
Sorry prof busy with class today. Will look at it later at night. Should be done by tomorrow.

**Comment by rd1992** [comment]
for new issue the window is given focus. commit [https://github.com/HubTurbo/HubTurbo/commit/f133b7bbeef2ac3a70d2af6d0489ff5432af88dc]
works on windows only for now. need to add error handling (in case window is closed etc.)

**Comment by damithc** [comment]
Need to complete at least full support for Windows (including error
handling) before we can merge this one. Mac support can be done as a
separate issue.

**Comment by rd1992** [comment]
yeah currently working on the error handling.

**Comment by rd1992** [comment]
done for windows here [https://github.com/HubTurbo/HubTurbo/commit/f75041c4e84da9d6207d04733ea251bb956b0186]
turning attention to osx now

**Comment by damithc** [comment]
Do Mac as a separate issue so that we can merge this one first?

**Comment by rd1992** [comment]
alright

**Comment by damithc** [comment]
Seems to work on Windows. Good job. Some rough edges yet to be worked out though.

**Comment by rd1992** [comment]
@damithc great ! could you tell me what the 'rough edges' are? that way when i finish with osx i can take care of them in one go

**Comment by damithc** [comment]
I've created separate issues for them.
